src/NHibernate/Engine/ActionQueue.cs

```diff
@@ -216,9 +216,9 @@ private void RegisterCleanupActions(IExecutable executable)
 				RegisterProcess(executable.AfterTransactionCompletionProcess);
 #pragma warning restore 618,619
 			}
-			if (executable.PropertySpaces != null)
+			if (executable is ICacheableExecutable cacheableExecutable && cacheableExecutable.QueryCacheSpaces != null)
```

Comment:
we need to have a fallback here in case someone uses a custom `IExecutable` implementation that does not implement the `ICacheableExecutable` interface.

Reply:
ok